### ISSUE TITLE
/en-US/docs/Learn/Server-side/Express_Nodejs/forms/Create_book_form

### ISSUE THREAD

**Issue by highpockets** [mdn/sprints#2852]
# Request type

- [ ] Please close this issue, I accidentally submitted it without adding any details
- [ ] New documentation
- [x] Correction or update

# Details

**EDIT: SEE LAST COMMENT WITH FIX**

The below screenshot shows the end of the book_form.pug template which dynamically checks checkboxes based on the value of `genre.checked`. This doesn't work, I got rid of the `checked` attribute altogether and replaced the value of the value attribute to `genre.checked` and the expected results displayed thereafter.

cheers

![Screen Shot 2020-02-26 at 12 56 13 PM](https://user-images.githubusercontent.com/3467943/75342842-7e455380-5897-11ea-95b1-815d9c57bfd3.png)

**Comment by highpockets** [comment]
Actually my fix introduced a new issue when submitting the form:

ValidationError: Book validation failed: genre: Cast to Array failed for value "" at path "genre"

**Comment by highpockets** [comment]
UPDATE:

The reason it wasn't working is because it seems if `checked` is visible, wether the value is 'true' or 'false', it shows the option as ticked on the form. So I put everything back to normal, but I included a condition to decide wether to display `checked`. I'm translating all PUG views to handlebars and I'm using chrome, but it shouldn't make a difference because in the PUG template, `checked` still gets used on every genre select box. I made the following handlebar helper and now everything works like a charm:

`hbs.handlebars.registerHelper('is_checked', function(checked){
    return (checked == 'true') ? new hbs.handlebars.SafeString(`checked`) : new hbs.handlebars.SafeString(``);
});`

and my input attributes look like:

`id={{this._id}} value={{this._id}} {{is_checked this.checked}}`

The approach with PUG will be different of course.
